code/modules/admin/player_panel/actions/punish.dm

```diff
@@ -152,3 +152,102 @@
+	action_tag = "reset_human_name"
+	name = "Reset Human Name"
+
+
+/datum/player_action/reset_human_name/act(var/client/user, var/mob/target, var/list/params)
+	var/mob/target_mob = target
+	if(!ismob(target_mob))
+		to_chat(user, SPAN_WARNING("[target.name] is not a mob!"))
+		return
+
+	if(isXeno(target_mob))
+		to_chat(user, SPAN_WARNING("[target.name] is a xeno!"))
+		return
+
+	if(!target_mob.ckey)
+		to_chat(user, SPAN_DANGER("Warning: Mob ckey for [target_mob.name] not found."))
+		return
+
+	if(alert(user, "Are you sure you want to reset name for [target_mob.ckey]?", , "Yes", "No") == "No")
+		return
+
+	var/new_name
+
+	if(alert(user, "Do you want to manually set the name for [target_mob.ckey]?", , "Yes", "No") == "No")
+		new_name = random_name(target_mob.gender)
+	else
+		var/raw_name = input(user, "Choose the new name:", "Name Input")  as text|null
+		if(!isnull(raw_name)) // Check to ensure that the user entered text (rather than cancel.)
+			new_name = reject_bad_name(raw_name)
+
+	if(!new_name)
+		to_chat(user, SPAN_NOTICE("Invalid name. The name should be at least 2 and at most [MAX_NAME_LEN] characters long. It may only contain the characters A-Z, a-z, -, ' and ."))
+		return
+
+	target_mob.change_real_name(target_mob, new_name)
+	if(istype(target_mob, /mob/living/carbon/human))
+		var/mob/living/carbon/human/target_human = target_mob
+		if(target_human.wear_id)
+			target_human.wear_id.name = "[target_human.real_name]'s ID Card"
+			target_human.wear_id.registered_name = "[target_human.real_name]"
+			if(target_human.wear_id.assignment)
+				target_human.wear_id.name += " ([target_human.wear_id.assignment])"
+
+	target_mob.client.prefs.real_name = new_name
+	target_mob.client.prefs.save_character()
+
+	message_staff("[user.ckey] has reset [target_mob.ckey] name")
+
+	to_chat(target_mob, SPAN_DANGER("Warning: Your name has been reset by [user.ckey]."))
+
+
+/datum/player_action/ban_human_name
+	action_tag = "ban_human_name"
+	name = "Ban Human Name"
+	permissions_required = R_BAN
+
+
+/datum/player_action/ban_human_name/act(var/client/user, var/mob/target, var/list/params)
+	if(!target.client || !target.ckey)
+		to_chat(user, SPAN_NOTICE("Target is lacking either client or ckey. Aborting."))
+		return
+
+	var/client/target_client = target.client
+
+	if(target_client.human_name_ban)
+		if(alert(user, "Are you sure you want to UNBAN [target.ckey] and let them use human names?", ,"Yes", "No") == "No")
+			return
+		target_client.human_name_ban = FALSE
+		target_client.prefs.human_name_ban = FALSE
+
+		target_client.prefs.save_preferences()
+		message_staff("[user.ckey] has unbanned [target.ckey] from using human names.")
+
+		notes_add(target.ckey, "Human Name Unbanned by [user.ckey]", user.mob)
+
+		to_chat(target, SPAN_DANGER("Warning: You can use human names again."))
+		return
+
+
+	if(alert("Are you sure you want to BAN [target.ckey] from ever using any human names?", , "Yes", "No") == "No")
+		return
+
```

Comment:
alert()s can take a long time

requires another client check and ckey check after